### PR TITLE
Updated credential-commons version

### PR DESCRIPTION
**Pull request by explicit-projects**

### CODE REVIEW

File: .babelrc [hunk 1]
```diff
@@ -1,14 +0,0 @@
-{
```
Comment (explicit-projects) [inline .babelrc:1]: The transform required for the transmute libraries ignores this file... it has been replaced with a `babel.config.js` equivalent instead

File: package.json [hunk 2]
```diff
@@ -1,13 +1,13 @@
 {
   "name": "@identity.com/dsr",
-  "version": "1.0.23",
+  "version": "3.0.0",
```
Comment (explicit-projects) [inline package.json:3]: Dunno why I skipped v2... considering it's already release as v3-beta, probably best to continue

File: package.json [hunk 3]
```diff
@@ -82,7 +82,21 @@
     "coverageDirectory": "reports/coverage",
     "collectCoverage": true,
     "testResultsProcessor": "./node_modules/jest-stare",
-    "verbose": true
+    "verbose": true,
+    "transformIgnorePatterns": [
```
Comment (explicit-projects) [inline package.json:86]: Needed as these modules don't work with jest without the transform

File: package.json [hunk 4]
```diff
@@ -1,13 +1,13 @@
...
-  "version": "1.0.23",
+  "version": "3.0.0",
   "description": "The Dynamic Scope Request (DSR) javascript library provides capability around securely requesting credential information between an ID Requester and an ID Holder",
   "main": "dist/cjs/index.js",
   "module": "dist/es/index.js",
   "browser": "dist/browser/index.js",
   "scripts": {
     "lint": "eslint ./src/**.js",
-    "test": "cross-env NODE_ENV=false jest",
+    "test": "cross-env NODE_ENV=test jest",
```
Comment (explicit-projects) [inline package.json:10]: Had to change the env for the transform to work correctly with jest

File: src/ScopeRequest.js [hunk 5]
```diff
@@ -311,7 +317,18 @@ class ScopeRequest {
     return true;
   }
 
-  constructor(uniqueId, requestedItems, channelsConfig, appConfig, partnerConfig, authentication = true, mode = 'ADVANCED') {
+  static async create(uniqueId, requestedItems, channelsConfig, appConfig, partnerConfig, authentication = true, mode = 'ADVANCED') {
```
Comment (explicit-projects) [inline src/ScopeRequest.js:320]: Static async method... some functions in the ScopeRequest constructor are now async, so moved them in here..

Comment (explicit-projects) [reply src/ScopeRequest.js:320]: This is the new way to create a scope request

File: src/ScopeRequest.js [hunk 6]
```diff
@@ -325,10 +342,7 @@ class ScopeRequest {
 
     this.timestamp = (new Date()).toISOString();
 
-    const credentialItems = [].concat(requestedItems);
-    if (ScopeRequest.validateCredentialItems(credentialItems)) {
```
Comment (explicit-projects) [inline src/ScopeRequest.js:329]: This is now async... moved to the static create above

File: src/resolver/Resolver.js [hunk 7]
```diff
@@ -1,5 +1,5 @@
 const sift = require('sift').default;
-const { definitions: ucaDefinitions } = require('@identity.com/uca');
+const { schemaLoader } = require('@identity.com/credential-commons');
```
Comment (explicit-projects) [inline src/resolver/Resolver.js:2]: no more direct dependancy on the uca lib... uca info accessed via CC

File: test/unit/services/configBranchPlatform.test.js [hunk 8]
```diff
@@ -1,6 +1,7 @@
+
+const originalPlatform = process.platform;
```
Comment (explicit-projects) [inline test/unit/services/configBranchPlatform.test.js:2]: this is weird... with the jest updates, using "this" no longer works... moved it to a variable instead

File: babel.config.js [hunk 9]
```diff
@@ -0,0 +1,17 @@
+module.exports = {
```
Comment (explicit-projects) [inline babel.config.js:1]: see coment in the deleted .babelrc file above

**Review by lucmir — APPROVED** [review]

**Pull request merged**